virtcontainers/sandbox.go

```diff
@@ -1433,3 +1433,18 @@ func (s *Sandbox) AddVhostUserDevice(devInfo api.VhostUserDevice, devType config
 	}
 	return fmt.Errorf("unsupported device type")
 }
+
+// Cleanup will clean up any dangling mounts created during a startup
+// procedure, as well as terminate the VM. This is used when a sandbox
+// failed during create.
+func (s *Sandbox) Cleanup() {
+	// stop VM
+	s.stop()
+	// unmount and delete mount dirs
+	bindUnmountAllRootfs(kataHostSharedDir, s)
+	os.RemoveAll(filepath.Join(kataHostSharedDir, s.id))
```

Comment:
`kataHostSharedDir` is specific to the `kata_agent.go` implementation, and should never be used by the generic  sandbox implementation. And about this, maybe something has been solved by #292

Reply:
@jamiehannaford I agree with @jshachm that you cannot use `Kata` specific paths from `sandbox.go`, this is specific to the agent implementation.
Talking about #292, I have just merged it !